### PR TITLE
add support for TLS client auth

### LINKED ISSUE #204: Is mTLS supported in this Grafana-Trino client?
On the Grafana datasources page for Trino, I see options on UI to add TLS certs but when I add them, it doesn't authenticate successfully. Is MTLS authentication supported or are those UI fields for which the functionality is yet to be implemented?

### PR DESCRIPTION
**Pull request by kenandominic**
resolves #204

### CODE REVIEW

File: pkg/trino/driver/driver.go [hunk 1]
```diff
@@ -35,11 +37,43 @@ func Open(settings models.TrinoDatasourceSettings) (*sql.DB, error) {
 			sslCertPath = f.Name()
 			defer os.Remove(f.Name())
 		}
+
+		if settings.Opts.TLS.ClientCertificate != "" {
+			if settings.Opts.TLS.ClientKey == "" {
+				return nil, errors.New("client certificate was configured without a client key")
+			}
+			clientCertF, certErr := os.CreateTemp("", "client-cert")
```
Comment (nineinchnick) [inline pkg/trino/driver/driver.go:45]: To get this plugin certified by Grafana, we can't use any IO here, so we can't create temporary files. In the latest `trino-go-client` version, we allowed passing the SSL cert as a string. See if we also need to make some changes to pass client certs like this.

Comment (nineinchnick) [reply pkg/trino/driver/driver.go:45]: Or can you use `X509KeyPair` instead of `LoadX509KeyPair`?

Comment (kenandominic) [reply pkg/trino/driver/driver.go:45]: thanks Jan, I have used `X509KeyPair` instead. I like that a custom http client can be provided to the trino go client as that allows for nice features like certificate reloading using `GetCertificate`.

File: pkg/trino/driver/driver.go [hunk 2]
```diff
@@ -17,14 +18,28 @@ const DriverName string = "trino"
...
+	if settings.Opts.TLS.ClientCertificate != "" {
+		if settings.Opts.TLS.ClientKey == "" {
+			return nil, errors.New("client certificate was configured without a client key")
+		}
+		var err error
+		clientCert, err = tls.X509KeyPair(
+			[]byte(settings.Opts.TLS.ClientCertificate),
+			[]byte(settings.Opts.TLS.ClientKey))
+		if err != nil {
+			return nil, errors.Join(errors.New("failed to load client certificate"), err)
```
Comment (nineinchnick) [inline pkg/trino/driver/driver.go:35]:
```suggestion
			return nil, fmt.Errorf("failed to load client certificate: %w", err)
```

File: pkg/trino/driver/driver.go [hunk 3]
```diff
@@ -17,14 +18,28 @@ const DriverName string = "trino"
...
+		}
+	}
 	client := &http.Client{
 		Transport: &http.Transport{
 			TLSClientConfig: &tls.Config{
 				InsecureSkipVerify: skipVerify,
+				Certificates:       []tls.Certificate{clientCert},
```
Comment (nineinchnick) [inline pkg/trino/driver/driver.go:42]: Is it ok to create a slice with a nil certificate? Maybe you should append to a slice after calling `tls.X509KeyPair()`.

Comment (kenandominic) [reply pkg/trino/driver/driver.go:42]: I can do that. however, from my manual testing agains our Trino server it didn't seem to cause any issues. I tested with just basic auth as well as just TLS client auth.

**Review by nineinchnick — APPROVED** [review]

**Pull request merged**